src/Axiom/Set/TotalMap.agda

```diff
@@ -1,34 +1,76 @@
 {-# OPTIONS --safe --no-import-sorts #-}
-{-# OPTIONS -v allTactics:100 #-}
 
-open import Agda.Primitive renaming (Set to Type)
-open import Axiom.Set
+open import Axiom.Set using ( Theory )
 
 module Axiom.Set.TotalMap (th : Theory) where
 
-open import Prelude
+open import Prelude hiding (lookup)
 
-open Theory th
-open import Axiom.Set.Map th
-open import Axiom.Set.Properties th
-open import Axiom.Set.Rel th
-open import Tactic.AnyOf
-open import Tactic.Defaults
+open import Agda.Primitive              using () renaming (Set to Type)
+open import Axiom.Set.Map th            using (left-unique; Map ; mapWithKey-uniq)
+open import Axiom.Set.Rel th            using (Rel ; dom ; dom∈)
+open import Interface.DecEq             using (DecEq ; _≟_)
+open import Relation.Nullary.Decidable  using (yes ; no)
+
+open Theory th    using (_∈_ ; map ; Set ; ∈-map ; ∈-map′)
+open Equivalence  using (to ; from)
 
 private variable A B : Type
 
--- Because of missing macro hygiene, we have to copy&paste this. https://github.com/agda/agda/issues/3819
-private macro
-  ∈⇒P = anyOfⁿᵗ (quote ∈-filter⁻' ∷ quote ∈-∪⁻ ∷ quote ∈-map⁻' ∷ quote ∈-fromList⁻ ∷ [])
-  P⇒∈ = anyOfⁿᵗ (quote ∈-filter⁺' ∷ quote ∈-∪⁺ ∷ quote ∈-map⁺' ∷ quote ∈-fromList⁺ ∷ [])
-  ∈⇔P = anyOfⁿᵗ (quote ∈-filter⁻' ∷ quote ∈-∪⁻ ∷ quote ∈-map⁻' ∷ quote ∈-fromList⁻ ∷ quote ∈-filter⁺' ∷ quote ∈-∪⁺ ∷ quote ∈-map⁺' ∷ quote ∈-fromList⁺ ∷ [])
 
 -- defines a total map for a given set
 total-map : Rel A B → Type
 total-map R = ∀ {a} → a ∈ map proj₁ R
 
-TotalMap : Type → Type → Type
-TotalMap A B = Σ (Rel A B) (λ R → left-unique R × total-map R)
 
-lookupMapᵗ : TotalMap A B → A → B
-lookupMapᵗ (_ , _ , tot) a = proj₂ (proj₁ (∈⇔P (tot {a})))
+record TotalMap (A B : Type) : Type where
+  field  rel : Set (A × B)
+         lun : left-unique rel
+         tot : ∀{a : A} → a ∈ dom rel
```

Comment:
Same with the name here, just name it `total-rel`.

Reply:
Yes, I agree the name `total` seems better than `total-map`.  Initially I went ahead and changed it, but then I changed it back in case you had a reason to prefer the latter. I'll change it to `total`.